src/test/java/data/DatabaseTest.java

```diff
@@ -0,0 +1,196 @@
+
+public class DatabaseTest {
+
+    ArrayList<Node> ndL = new ArrayList<Node>();
+    ArrayList<Provider> pvdL = new ArrayList<Provider>();
+    ArrayList<Floor> flL = new ArrayList<Floor>();
+
+    @Before
+    public void setUp(){
+        ndL.clear();
+        pvdL.clear();
+        flL.clear();
+        DatabaseController.createConnection();
+
+        DatabaseController.droptablesForShittyTesting();
+        DatabaseController.initializeProviderTable();
+        DatabaseController.initializeFloorTable();
+        DatabaseController.initializeNodeTable();
+        DatabaseController.initializeOfficeTable();
+        DatabaseController.initializeNeighborTable();
+
+        Floor fl1 = new Floor(01, "TestFloor", 1);
+        ConcreteNode n001 = new ConcreteNode(001, new ArrayList<>(Arrays.asList("Room001", "room")), 1, 1, fl1);
+        ConcreteNode n002 = new ConcreteNode(002, new ArrayList<>(Arrays.asList("Room002", "room")), 2, 2, fl1);
+        ConcreteNode n003 = new ConcreteNode(003, new ArrayList<>(Arrays.asList("Room003", "room")), 3, 3, fl1);
+        Provider p111 = new Provider(111, "Donald", "Trump");
+        Provider p222 = new Provider(222, "Barack", "Obama");
+        n001.addNeighbor(n002);
+        n002.addNeighbor(n001);
+        p111.addLocation(n003);
+        flL.add(fl1);
+        ndL.add(n001);
+        ndL.add(n002);
+        ndL.add(n003);
+        pvdL.add(p111);
+        pvdL.add(p222);
+
+        testInsert();
+    }
+
+    @After
+    public void shutdown(){
+        DatabaseController.shutdown();
+    }
+
+    @Test
+    public void testInsert(){
+        DatabaseController.insertFloor(01, "TestFloor", 1);
+        DatabaseController.insertNode(001, "Room001", "room", 1, 1, 01);
+        DatabaseController.insertNode(002, "Room002", "room", 2, 2, 01);
+        DatabaseController.insertNode(003, "Room003", "room", 3, 3, 01);
+        DatabaseController.insertProvider(111, "Donald", "Trump");
+        DatabaseController.insertProvider(222, "Barack", "Obama");
+        DatabaseController.insertNeighbor(001, 002);
+        DatabaseController.insertNeighbor(002, 001);
+        DatabaseController.insertOffice(111, 003);
+
+    }
+
+    @Test
+    public void testInitialization(){
+
+        DatabaseController.droptablesForShittyTesting();
+
+        DatabaseController.initializeProviderTable();
+        DatabaseController.initializeFloorTable();
+        DatabaseController.initializeNodeTable();
+        DatabaseController.initializeOfficeTable();
+        DatabaseController.initializeNeighborTable();
+        testInsert();
+        DatabaseController.initializeAllFloors();
+        DatabaseController.initializeAllNodes();
+        DatabaseController.initializeAllProviders();
+
+        for(int i=0;i<flL.size();i++){
+            compareFloor(flL.get(i), DatabaseController.getFloorList().get(i));
+        }
+
+        //for(int i=0;i<ndL.size();i++){
+        //TODO: Why is data coming in this order?
```

Comment:
This TODO doesn't affect any real data as far as i can tell, unless we do something that relies on nodes being sorted in a certain way in nodeList (which we really shouldn't do)